pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -119,32 +132,57 @@ def __init__(
         self.min_num_params = min_num_params
         self.state_dict_device = torch.device("cpu") if state_dict_to_cpu else None
         self._process_group = None
+        self._pids: Optional[List[int]] = None
+        self._sync_dir: Optional[str] = None
+        self._rank_0_will_call_children_scripts: bool = False
+        self.set_world_ranks()
+
+    @property
+    def is_distributed(self) -> bool:
+        return True
 
     @property
     def process_group(self):
         if self._process_group is None:
             self._process_group = torch.distributed.new_group()
         return self._process_group
 
-    def setup_distributed(self) -> None:
+    @property
+    def root_device(self) -> torch.device:
+        return self.parallel_devices[self.local_rank]
+
+    @property
+    def num_nodes(self) -> int:
+        return self._num_nodes
+
+    @num_nodes.setter
+    def num_nodes(self, num_nodes: int) -> None:
+        # note that world ranks is related to num_nodes, when resetting it, need to reset world ranks
+        self._num_nodes = num_nodes
+        self.set_world_ranks()
+
+    @property
+    def num_processes(self) -> int:
+        return len(self.parallel_devices) if self.parallel_devices is not None else 0
+
+    def setup_environment(self) -> None:
```

Comment:
Let's move this code into `setup_distributed` as DDP. Could maybe be moved to the Parallel class.